packages/teams-js/src/private/store.ts

```diff
@@ -150,6 +162,7 @@ export async function openStoreExperience(openStoreParams: OpenStoreParams): Pro
       openStoreParams.dialogType,
       (openStoreParams as OpenAppDetailParams).appId,
       (openStoreParams as OpenSpecificStoreParams).collectionId,
```

Comment:
this function should be validating that the size being passed in is legal values. Off the top of my head, this is probably at least that neither size is < 0, but you may want to put other restrictions on it such as more reasonable minimums and maximums. Without that a dev could ask you to open a 10000x10000px dialog

Reply:
I've gone through the whole flow of dialog service and the dialog size will be handled in the implementation of dialog service on each hub/host. Let's take Harmony as an example.
```
// this is the implementation of dialog service in Harmony
const TaskModuleDialogControl = (size: DialogSize) => {
        // windowHeight is the height of current browser window
        const { height: windowHeight, width: windowWidth } =
            useSingletonWindowSize();

        const [sizes, setSizes] = useState<DialogSize>({
            height: getHeight(windowHeight, size.height),
            width: getWidth(windowWidth, size.width),
        });
}
export const getHeight = (
    windowHeight: number,
    height: number | DialogDimension | undefined,
): number | DialogDimension => {
    if (!height) return DialogDimension.Medium;
    if (typeof height === 'number') {
        return clampDimension(windowHeight, height);
    }

    switch (height as DialogDimension) {
        case DialogDimension.Small:
            return windowHeight * 0.2;
        case DialogDimension.Medium:
            return windowHeight * 0.5;
        case DialogDimension.Large:
            return windowHeight * 0.66;
    }
};
```

And when implementing the openStore api in storeService, it will finally call and pass the size info to dialog.open() in dialog service, which means the dialog service in each host will take care of the size info in the end, no matter the size is number out of bound or an abstract size.
```
export class StoreService implements IStoreService {
  constructor(protected dialogService: IDialogService) {}

  public async openStoreExperience(params: OpenStoreParams): Promise<OpenStoreResults> {
    const { dialogType, size} = params;
    const url = getUrlFromDialogType(dialogType);
    const dialogInfo: DialogWithUrl = {
      url: urlFormatted,
      type: DialogContentType.Url,
    };
    await this.dialogService!.urlService!.open(dialogInfo);
  }
}
```

Reply:
We can certainly let hosts decide what to cap dialog sizes at, although I worry a little bit that many of them might not think of that case and could run into bugs where they have to cap the size because someone asked for one that was too large.

It *does* seem though that teams-js should at least validate that the dialog size is > 0. It won't ever be valid to ask for a dialog that is 0px high or wide, and if we catch that in teams-js we can fail faster and let the developer know that the dialog won't be opening asap without them having to wait for a round trip to the host.

Reply:
added unit test & e2e test